keps/sig-node/5246-cgroup-cpu-share-to-weight-conversion/README.md

```diff
@@ -0,0 +1,939 @@
+keeping the values in the same percentile in this range.
+The second one focuses on aligning the values to the default CPU weight of `100` for a container requesting 1 CPU.
+Accordingly, 512m CPU will result in a value of `50`, which aligns with the default v1 CPU shares of `512` for a container requesting 1 CPU.
+
+Let cpu.shares be 1024. Therefore: `1 + ((cpu.shares – 2) × 99) / (1024 – 2) == 100`.
+
+Let cpu.shares be 512. Therefore: `1 + ((cpu.shares – 2) × 99) / (1024 – 2) == 50.4031 ~= 50`.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a Kubernetes user, when I used to work with v1, a container asking for 1 CPU had the same CPU shares as a non-kubernetes
+workload. After moving to cgroup v2 I expect this behavior to stay the same, but instead,
+the CPU priority for Kubernetes workloads is much lower than non-kubernetes workloads compared to v1.
+
+#### Story 2
+As a Kubernetes user, I want to be able to configure sub-cgroups inside a container to further
+distribute resources inside the container. While I could do that nicely with v1, with v2 the granularity is not fine-grained enough.
+As an example, `100m` CPU on v2 results with `4` CPU weight, while on v1 it would be `102` shares.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- A significant amount of the work would need to land in other layers, mainly OCI runtimes and the CRI.
+- We'll probably need a CRI configuration to ensure coordination between the CRI and the OCI runtimes implementations,
+and to ensure it lands at the same version, as suggested
+[here](https://github.com/kubernetes/kubernetes/issues/131216#issuecomment-2806656165).
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+The main risk comes from the fact that we use the same formula for a decent amount of time, hence there's always
+a risk that a user relies on the exact values that we're using.
+
+That being said, the formula in entirely an implementation detail that's most probably not being counted
+to have certain concrete values. In any way, we should ensure that the new formula is well documented
+and that the change is properly communicated to the users.
```

Comment:
Will there be a way for the users to configure back to the previous behavior, in the case of unexpected issues resulted from the change?

Reply:
I don't think this mechanism will be implemented by the OCI runtimes, the entire mechanism is a workaround for using cgroup v1 settings in a cgroup v2 environment.

If someone wants to have full control of the cgroup v2 values, then the must use the native cgroupv2 unified map to pass the correct value down the stack

Reply:
I mainly asked because if we simply switched the implementation, it may have unexpected impact on user workloads. Having an option to preserve the original behavior can be important

Reply:
If the change was purely at the OCI/CRI level, I'd say we can delegate this decision to these layers.
However, since there's a little k8s code that's also effected (e.g. pod-level resources) if a user would want to roll-back to the previous behavior we would need such a configurable.

Saying that, I'm not entirely sure it will be valuable and that it's worth introducing a config that will most likely be deprecated very quickly.

In the end of the day, the only users I can think of that will be negatively affected by this are users who expect concrete CPU weight values that would now be changed. I think this use-case is very rare if exists at all. In any case, I'm open to this approach.